sktime/forecasting/base/_adapters.py

```diff
@@ -63,7 +63,7 @@ def _instantiate_model(self):
         )
         return self
 
-    def _get_sp(sp):
+    def _get_sp(self, sp):
```

Comment:
Wouldn't make this a method but a simple standalone function something like `_check_sp`. We also have a `check_sp` function [here](https://github.com/alan-turing-institute/sktime/blob/28646ab32d083f799d7723efcf514ee558c43694/sktime/utils/validation/forecasting.py#L151) which you could use in `fit`